File: backend/src/main/scala/kpn/api/common/monitor/MonitorRouteChangesPage.scala
package kpn.api.common.monitor

case class MonitorRouteChangesPage(
  routeId: String,
  routeName: String,
  groupName: String,
  groupDescription: String,
  impact: Boolean,
  pageSize: Long,
  pageIndex: Long,
  totalChangeCount: Long,
  changes: Seq[MonitorRouteChangeSummary]
)

File: backend/api/src/main/java/kpn/api/common/monitor/MonitorRouteChangesPage.java
package kpn.api.common.monitor;

import kpn.api.common.monitor.MonitorRouteChangeSummary;

import com.google.common.collect.ImmutableList;

public record MonitorRouteChangesPage(
  String routeId,
  String routeName,
  String groupName,
  String groupDescription,
  Boolean impact,
  Long pageSize,
  Long pageIndex,
  Long totalChangeCount,
  ImmutableList<MonitorRouteChangeSummary> changes
) {
}
